src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ModelFactoryAnalyzer.cs

```diff
@@ -0,0 +1,212 @@
+                }
+            }
+        }
+
+        private static bool IsClientType(INamedTypeSymbol namedType)
+        {
+            return namedType.TypeKind == TypeKind.Class &&
+                   namedType.Name.EndsWith(ClientSuffix) &&
+                   namedType.DeclaredAccessibility == Accessibility.Public;
+        }
+
+        private static bool IsModelFactoryType(INamedTypeSymbol namedType)
+        {
+            return namedType.TypeKind == TypeKind.Class &&
+                   namedType.Name.EndsWith(ModelFactorySuffix) &&
+                   namedType.IsStatic &&
+                   namedType.DeclaredAccessibility == Accessibility.Public;
+        }
+
+        private static void ExtractOutputModelsFromClientType(INamedTypeSymbol clientType, HashSet<ITypeSymbol> outputModels)
+        {
+            foreach (var member in clientType.GetMembers())
+            {
+                if (member is IMethodSymbol method && method.DeclaredAccessibility == Accessibility.Public)
+                {
+                    // Skip property accessors as they are not client methods
+                    if (method.AssociatedSymbol is IPropertySymbol)
+                        continue;
+
+                    var outputModel = ExtractOutputModelFromReturnType(method.ReturnType);
+                    if (outputModel != null)
+                    {
+                        outputModels.Add(outputModel);
+                    }
+                }
+            }
+        }
+
+        private static void ExtractReturnTypesFromModelFactory(INamedTypeSymbol factoryType, HashSet<ITypeSymbol> modelFactoryMethods)
+        {
+            foreach (var member in factoryType.GetMembers())
+            {
+                if (member is IMethodSymbol method && 
+                    method.DeclaredAccessibility == Accessibility.Public && 
+                    method.IsStatic)
+                {
+                    // Add the return type of the factory method
+                    modelFactoryMethods.Add(method.ReturnType);
+                }
+            }
+        }
+
+        private static ITypeSymbol ExtractOutputModelFromReturnType(ITypeSymbol returnType)
+        {
+            ITypeSymbol unwrappedType = returnType;
+
+            // Unwrap Task<T>
+            if (returnType is INamedTypeSymbol namedType && 
+                namedType.IsGenericType && 
+                namedType.Name == TaskTypeName)
+            {
+                unwrappedType = namedType.TypeArguments.FirstOrDefault();
+                if (unwrappedType == null) return null;
+            }
+
+            ITypeSymbol modelType = null;
+
+            // Check for Azure client method return types and extract the model type
+            if (IsOrImplements(unwrappedType, ResponseTypeName, AzureNamespace) ||
+                IsOrImplements(unwrappedType, NullableResponseTypeName, AzureNamespace) ||
+                IsOrImplements(unwrappedType, ClientResultTypeName, AzureNamespace) ||
```

Comment:
client result is not in Azure namespace. It's in System.ClientModel. Same with some other types in this list